configure.ac

```diff
@@ -7403,8 +7403,7 @@ AC_DEFINE([PY_SSL_DEFAULT_CIPHERS], [1])
 
 # builtin hash modules
 default_hashlib_hashes="md5,sha1,sha2,sha3,blake2"
-AC_DEFINE([PY_BUILTIN_HASHLIB_HASHES], [], [enabled builtin hash modules]
-)
+AC_DEFINE_UNQUOTED([PY_BUILTIN_HASHLIB_HASHES], ["$with_builtin_hashlib_hashes"], [enabled builtin hash modules])
```

Comment:
This will not work; the variable `with_builtin_hashlib_hashes` is not defined yet.

Reply:
I'm not really sure how to fix it then -- all I know is that the result of the previous style is two #defines in the conftest.h file... are you familiar with autoconf?

Reply:
(Oh and this change *did* fix my particular issue and seems to give the expected result, but I appreciate that this may not be the right way to go about it.)

Reply:
When changing Autoconf input, also check the generated output. You're now writing `$with_builtin_hashlib_hashes` to `confdefs.h` before `with_builtin_hashlib_hashes` is set (it's set in the `AC_ARG_WITH` a few lines below).

The correct solution is to keep only the latter `AC_DEFINE_UNQUOTED` in L7423 and remove the one in L7406.

> Oh and this change did fix my particular issue and seems to give the expected result,

I suspect you're not getting the expected result. The expected result is a CSV C string literal with a list of the buildin hashes, but the actual result of this PR is that the macro is defined to the empty string:

```sh
$ ./configure
$ grep PY_BUILTIN_HASHLIB_HASHES pyconfig.h                  
#define PY_BUILTIN_HASHLIB_HASHES ""
```

My expected result would be:

```sh
$ ./configure
$ grep PY_BUILTIN_HASHLIB_HASHES pyconfig.h
#define PY_BUILTIN_HASHLIB_HASHES "md5,sha1,sha2,sha3,blake2"
```

Reply:
Yeah you are correct, this was definitely not working as intended. Thanks, this is now fixed.